model/src/main/kotlin/licenses/ResolvedLicense.kt

```diff
@@ -121,19 +121,20 @@ data class ResolvedLicense(
 }
 
 private fun Collection<ResolvedCopyrightFinding>.toResolvedCopyrights(process: Boolean): List<ResolvedCopyright> {
-    val allStatements = map { it.statement }
-    val processedStatements = if (process) {
-        CopyrightStatementsProcessor().process(allStatements).toMap()
-    } else {
-        allStatements.associateBy({ it }, { setOf(it) })
+    if (!process) return map { ResolvedCopyright(it.statement, setOf(it)) }
+
+    val statementToFinding = associateBy { it.statement }
+    val (processedStatements, unprocessedStatements) = CopyrightStatementsProcessor().process(statementToFinding.keys)
+
+    val processedCopyrights = processedStatements.mapNotNull { (statement, originals) ->
```

Comment:
Was you mention optimizations, was this function too slow?

Reply:
> was this function too slow?

I did not measure it. It just occurred to me from the code that for some cases too much work was being done, like `in`-checks in a double-nested loop.